sdk-src/src/test/java/com/hpe/adm/nga/sdk/network/google/TestGoogleHttpClient.java

```diff
@@ -96,6 +99,31 @@ public void testRequestRetry() throws Exception {
                 .invoke("execute", any(), anyInt());
     }
 
+    @Test
+    public void testCustomSettings() {
+        int connTimeout = 2345;
+
+        Octane.OctaneCustomSettings settings = new Octane.OctaneCustomSettings() {{
+            readTimeout = 120000;
+            connectionTimeout = 2345;
+        }};
+
+        GoogleHttpClient client = new GoogleHttpClient("http://google.com:8090", settings );
+        OctaneHttpRequest request = new OctaneHttpRequest.GetOctaneHttpRequest("http://google.com:9090");
+
+        long start = System.currentTimeMillis();
+        try {
+            client.execute(request);
+        } catch (Exception e) {
+            long end = System.currentTimeMillis();
+
+            Assert.assertTrue(e.getCause() instanceof SocketTimeoutException);
+            long duration = end - start;
+            System.out.println("Error after: "  + (end - start));
+            Assert.assertTrue(duration < 3000 && duration > 2000);
```

Comment:
All we know about this exception is that it should take more than settings.connectionTimout to complete. If the duration is over 3 seconds, that does not necessarily mean that there is a problem with the code. I would only assert that the duration is > than the settings.connectionTimout  and maybe that it`s less than the default connection timeout.

Reply:
indeed